sale_disable_inventory_check/models/product_template.py

```diff
@@ -0,0 +1,22 @@
+# Copyright Komit <http://komit-consulting.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    check_stock_on_sale = fields.Boolean(
+        company_dependent=True,
+        help="This checkbox disable warning 'Not enough inventory'"
+             " when there isn't enough product in stock",
+        string="Disable Check Stock Message",
+        default=True,
+    )
+
+    @api.multi
+    def _check_stock_on_sale(self):
+        return all((
+            self.check_stock_on_sale,
+            self.categ_id.check_stock_on_sale, True))
```

Comment:
Having a fallback on the category value works less well for boolean fields. It is different for Many2one fields because in that case, the option on the product is either entered in which case it is used, or it is not entered in which case there is a fallback on the category setting. In the case of boolean fields, how is the user to know when the fallback on the category value kicks in?

Reply:
Yeah, this is a good point. If you want to have the field defer to the category it should maybe be a `Selection` field that enables you to explicitly choose between:
- Skip the check
- Do not skip the check
- Defer to the setting on the category

The field might be allowed to be left empty as well, but then you need to decide what that means. If it means "do nothing different", then that's the same as "Do not skip the check".

Reply:
i will take into account this comment https://github.com/OCA/sale-workflow/pull/805#discussion_r273815740